features/netsocket/TCPSocket.cpp

```diff
@@ -19,8 +19,7 @@
 #include "mbed_assert.h"
 
 TCPSocket::TCPSocket()
-    : _pending(0), _read_sem(0), _write_sem(0),
-      _read_in_progress(false), _write_in_progress(false)
+    : _pending(0), _read_sem(0), _write_sem(0)
```

Comment:
Consider the example in #3058. I don't follow how this assertion is needed for tcp or discourages bugs. Let's just continue this conversation in this pr's body